### PR TITLE
samples: usb: hid-cdc: Convert to use gpio_dt_spec

### PR DESCRIPTION
**Pull request by galak**
Move sample to use gpio_dt_spec for GPIO access.

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

File: samples/subsys/usb/hid-cdc/src/main.c [hunk 1]
```diff
@@ -20,38 +20,25 @@ LOG_MODULE_REGISTER(main);
...
+static const struct gpio_dt_spec sw1_gpio = GPIO_DT_SPEC_GET(SW1_NODE, gpios);
 #endif
 
 #define SW2_NODE DT_ALIAS(sw2)
 
 #if DT_NODE_HAS_STATUS(SW2_NODE, okay)
-#define PORT2		DT_GPIO_LABEL(SW2_NODE, gpios)
-#define PIN2		DT_GPIO_PIN(SW2_NODE, gpios)
-#define PIN2_FLAGS	DT_GPIO_FLAGS(SW2_NODE, gpios)
+static const struct gpio_dt_spec sw2_gpio = GPIO_DT_SPEC_GET(SW1_NODE, gpios);
```
Comment (jfischer-no) [inline samples/subsys/usb/hid-cdc/src/main.c:35]: `static const struct gpio_dt_spec sw2_gpio = GPIO_DT_SPEC_GET(SW2_NODE, gpios);`

File: samples/subsys/usb/hid-cdc/src/main.c [hunk 2]
```diff
@@ -20,38 +20,25 @@ LOG_MODULE_REGISTER(main);
...
 #endif
 
 #define SW3_NODE DT_ALIAS(sw3)
 
 #if DT_NODE_HAS_STATUS(SW3_NODE, okay)
-#define PORT3		DT_GPIO_LABEL(SW3_NODE, gpios)
-#define PIN3		DT_GPIO_PIN(SW3_NODE, gpios)
-#define PIN3_FLAGS	DT_GPIO_FLAGS(SW3_NODE, gpios)
+static const struct gpio_dt_spec sw3_gpio = GPIO_DT_SPEC_GET(SW1_NODE, gpios);
```
Comment (jfischer-no) [inline samples/subsys/usb/hid-cdc/src/main.c:41]: `static const struct gpio_dt_spec sw3_gpio = GPIO_DT_SPEC_GET(SW3_NODE, gpios);`

File: samples/subsys/usb/hid-cdc/src/main.c (hunk 1 above)
Comment (galak) [reply samples/subsys/usb/hid-cdc/src/main.c]: good catch, thanks.

File: samples/subsys/usb/hid-cdc/src/main.c (hunk 2 above)
Comment (galak) [reply samples/subsys/usb/hid-cdc/src/main.c]: good catch, thanks.

**Review by jfischer-no — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**